src/Common/PageCache.cpp

```diff
@@ -150,7 +150,7 @@ void PageCache::autoResize(size_t memory_usage, size_t memory_limit)
     size_t peak;
     {
         std::lock_guard lock(mutex);
-        size_t usage_excluding_cache = memory_usage - std::min(cache_size, memory_usage);
+        size_t usage_excluding_cache = memory_usage - std::min(cache_size, size_t(std::max(memory_usage, Int64(0))));
```

Comment:
`size_t usage_excluding_cache`: unsigned underflow if `memory_usage` is negative.
Should it be signed?